libs/execution_context/ebpf_native.c

```diff
@@ -468,6 +647,97 @@ _ebpf_native_release_reference_internal(void* base_object, ebpf_file_id_t file_i
     _ebpf_native_release_reference(base_object);
 }
 
+static ebpf_result_t
+_ebpf_native_validate_native_module(_In_opt_ const metadata_table_t* table, _In_ const GUID* client_module_id)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    metadata_table_t local_table = {0};
+    bpf2c_version_t version;
+    program_entry_t* programs = NULL;
+    size_t program_count = 0;
+    map_entry_t* maps = NULL;
+    size_t map_count = 0;
+    map_initial_values_t* map_initial_values = NULL;
+    size_t map_initial_values_count = 0;
+    global_variable_section_info_t* global_variables = NULL;
+    size_t global_variable_count = 0;
+
+    if (!table || !table->programs || !table->maps) {
+        EBPF_LOG_MESSAGE_GUID(
+            EBPF_TRACELOG_LEVEL_ERROR,
+            EBPF_TRACELOG_KEYWORD_NATIVE,
+            "The metadata table size is invalid",
```

Comment:
suggest: Do we want to have a slightly different error statement than the below, to allow us to differentiate these two cases? Alternatively, should we just convert the below into the same if check as this one?

Reply:
fixed.